examples/bridge-app/linux/UserInputBackend.cpp

```diff
@@ -0,0 +1,328 @@
+        return;
+    }
+    EmberAfDeviceType type = { (uint16_t) atoi(tokens[1].c_str()), 1u };
+    if (tokens.size() >= 3)
+    {
+        type.deviceVersion = (uint8_t) atoi(tokens[2].c_str());
+    }
+    printf("Adding device type %d ver %d\n", type.deviceId, type.deviceVersion);
+
+    g_pending->AddDeviceType(type);
+}
+
+bool FindIndexAttr(const std::vector<std::string> & tokens, ClusterImpl ** cluster, chip::AttributeId * attrId)
+{
+    if (tokens.size() < 4)
+        return false;
+    int index = atoi(tokens[1].c_str());
+    DynamicDeviceImpl * dev;
+    if (index == -1)
+    {
+        dev = g_pending.get();
+    }
+    else if ((uint32_t) index < g_device_impls.size())
+    {
+        dev = g_device_impls[index].get();
+    }
+    else
+    {
+        return false;
+    }
+    *cluster                  = nullptr;
+    chip::ClusterId clusterId = (chip::ClusterId) atoi(tokens[2].c_str());
+    for (auto * c : dev->clusters())
+    {
+        if (c->GetClusterId() == clusterId)
+        {
+            *cluster = c;
+            break;
+        }
+    }
+    if (!*cluster)
+        return false;
+    *attrId = (chip::AttributeId) atoi(tokens[3].c_str());
+    return true;
+}
+
+void ParseValue(std::vector<uint8_t> * data, uint16_t size, const std::string & str, EmberAfAttributeType type)
+{
+    switch (type)
+    {
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE:
+        if (str.size() >= 255)
+            return;
+        data->resize(size);
+        memcpy(data->data() + 1, str.data(), str.size());
+        data->data()[0] = (uint8_t) str.size();
+        break;
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE: {
+        if (str.size() >= 65534)
+            return;
+        data->resize(size);
+        memcpy(data->data() + 2, str.data(), str.size());
+        uint16_t length = (uint16_t) str.size();
+        memcpy(data->data(), &length, 2);
+        data->data()[0] = (uint8_t) str.size();
+    }
+    break;
+    case ZCL_STRUCT_ATTRIBUTE_TYPE:
+        // Writing structs not supported yet
+        break;
+    case ZCL_SINGLE_ATTRIBUTE_TYPE: {
+        float v = (float) atof(str.c_str());
+        data->resize(size);
+        memcpy(data->data(), &v, size);
+    }
+    break;
+    case ZCL_DOUBLE_ATTRIBUTE_TYPE: {
+        double v = atof(str.c_str());
+        data->resize(size);
+        memcpy(data->data(), &v, size);
+    }
+    break;
+    default: {
+        // Assume integer
+        uint64_t v = strtoull(str.c_str(), nullptr, 10);
+        data->resize(size);
+        memcpy(data->data(), &v, size);
+    }
+    break;
+    }
+}
+
+void SetValue(const std::vector<std::string> & tokens)
+{
+    ClusterImpl * cluster;
+    chip::AttributeId attrId;
+    if (!FindIndexAttr(tokens, &cluster, &attrId))
+    {
+        printf("Invalid device or cluster\n");
+        return;
+    }
+    if (tokens.size() < 5)
```

Comment:
All methods here need comments and printfs should have some form of 'help'. What are the 5 tokens expected here?